qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,690 @@
+                mapped_params = []
+                for series_param in series_params:
+                    try:
+                        param_idx = self.__param_names__.index(series_param)
+                    except ValueError as ex:
+                        raise QiskitError(
+                            f"Local function parameter {series_param} is not defined in "
+                            f"this class. {series_param} not in {self.__param_names__}."
+                        ) from ex
+                    mapped_params.append(params[param_idx])
+
+                # find fit function
+                f_index = curve_properties.fit_func_index
+                try:
+                    return self.__fit_funcs__[f_index](xvals, *mapped_params)
+                except IndexError as ex:
+                    raise QiskitError(
+                        f"Fit function of index {f_index} is not defined."
+                    ) from ex
+
+        raise QiskitError(f"A curve {curve_name} is not defined in this class.")
+
+    def _run_analysis(self, experiment_data: ExperimentData, **options):
+        """Run analysis on circuit data.
+
+        Args:
+            experiment_data: the experiment data to analyze.
+            options: kwarg options for analysis function.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        analysis_result = AnalysisResult()
+
+        # Extract curve entries from experiment data
+        try:
+            curve_data = self._extract_curves(experiment_data)
+            analysis_result["raw_data"] = curve_data
+        except DataProcessorError as ex:
+            analysis_result["error_message"] = str(ex)
+            analysis_result["success"] = False
+            return analysis_result, list()
+
+        # Run fitting
+        # pylint: disable=broad-except
+        try:
+            fit_data = self._run_fitting(curve_data=curve_data, **options)
+            analysis_result.update(fit_data)
+            analysis_result["success"] = True
+        except Exception as ex:
+            analysis_result["error_message"] = str(ex)
+            analysis_result["success"] = False
+
+        # Post-process analysis data
+        analysis_result = self._post_processing(analysis_result)
+
+        # Create figures
+        figures = self._create_figure(curve_data=curve_data, fit_data=analysis_result)
+
+        # Store raw data
+        raw_data = dict()
+        for datum in curve_data:
+            raw_data[datum.curve_name] = {
+                "x_values": datum.x_values,
+                "y_values": datum.y_values,
+                "y_sigmas": datum.y_sigmas,
+            }
+        analysis_result["raw_data"] = raw_data
+
+        return analysis_result, figures
```

Comment:
You gave examples of how T1, Ramsey, and IRB can be done without overriding methods. The T1 and Ramsey experiments handle delay times given in a unit chosen by the user, and return the result in SI (seconds for T1, Hz for Ramsey), while providing information about the units in the analysis result. Do the bases classes support unit handling?

Reply:
Similarly, according to the current form of Terra's PR 5499, every experiment, in addition to the fit results, must provide a single golden value (together with its unit and stderr, separately from all the other fitted parameters). Is this reflected here? Would you suggest to extract it in post processing? I think this should be supported by the base classes (if not already), for example that the base `_post_processing` method receives the name of the prestigious parameter. For example, for T1 this would be `__important_parameter__ = "tau"`.

Reply:
Unit conversion is not implemented here. This can be handled in `_run_fitting` in each subclass (the `curve_data` has metadata extracted from circuit results, thus you can get unit from there). If most of curve analysis subclasses have unit dependency it's worth implementing it -- in this case we need to make `unit` a special field in the metadata.

Thanks for telling me the prestigious parameter. Sounds like this should be hard-corded in the class attribute as you said.